network-api/networkapi/wagtailpages/pagemodels/publications/article.py

```diff
@@ -88,11 +102,47 @@ class ArticlePage(FoundationMetadataPageMixin, Page):
         related_name='+',
     )
 
+    hero_layout = models.CharField(
+        max_length=25,
+        choices=[
+            ('full_screen', 'Full Screen'),
+            ('image_left', 'Image Left'),
+            ('image_right', 'Image Right'),
+            ('static', 'Static'),
+        ],
+        default='static',
+    )
+
+    download_button_style = models.CharField(
+        max_length=25,
+        choices=[
+            ('primary', 'Primary'),
+            ('secondary', 'Secondary'),
+            ('tertiary', 'Tertiary'),
+        ],
+        default='primary',
+    )
+
+    download_button_icon = models.ForeignKey(
+        'wagtaildocs.Document',
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        related_name='+',
+        help_text="Custom Icon for download button, please use http://feathericons.com"
```

Comment:
`https`, at the very least, but I'm a little confused why a download icon is a foreignkey to a Document, what are we actually doing here?

Reply:
Hi @Pomax, 

The request from design was to have users be able to update the page's download button with a custom icon from feathericons.com.

FeatherIcons are only downloadable in SVG which appear to be incompatible with the imageField, which I believe would have been the ideal use case.

After some further investigation, I found [this comment ](https://www.reddit.com/r/WagtailCMS/comments/eey2wn/comment/fh6jtvu/?utm_source=share&utm_medium=web2x&context=3)from Kalob where he explains the limitations of SVG and the wagtail ImageField, and a workaround in order to upload them to the CMS. 

> SVG's aren't technically images, they're xml-like files what get read and rendered by a browser (or other design program like Photoshop or Illustrator).
> Because they are files, they need to be uploaded as a file, not an image.
> In Wagtail this goes through either a `models.FileField()` Page property, or through the document uploader and manager.
```
from wagtail.core.models import Page
from wagtail.documents.edit_handlers import DocumentChooserPanel 

class YourPage(Page):
    svg = models.ForeignKey(
        'wagtaildocs.Document',
        blank=True, # or False 
        null=True, # or False 
        related_name='+',
        on_delete=modules.SET_NULL, # Only works with null=True
    )

    content_panels = Page.content_panels + [
        DocumentChooserPanel("svg"),
    ]
```



From Kalob's explanation, it appears that there 2 ways to be able to achieve this:

1. The first one being to use the document manager as we are here, which in my thinking was the most similar to using the image manager.
2. The second option Kalob mentioned was using a filefield, which come to think of it, might be a better option as that would provide a direct file selector instead of using the document manager. For example, here is what the FileField looks like in the "News" Model's thumbnail field:
<img width="1001" alt="Screen Shot 2022-03-22 at 5 41 35 PM" src="https://user-images.githubusercontent.com/18314510/159599812-5aed8da3-a334-49e5-b743-56418316f264.png">

A possible third option, would be to use the [wagtailSVG](https://pypi.org/project/wagtailsvg/) package, the last update to it appears to be Feb 2022, however it also seems to be a small project with 26 stars.  

Let me know if that helped answer the question, or if you would like me to implement this a different way 😄 thanks!

Reply:
Ah, that makes sense. It would probably be good to explain why a Document is used in a code comment here.